src/types/binary/binaryType.ts

```diff
@@ -4,18 +4,20 @@ export interface IByte {
   readonly value: number
 }
 
-export abstract class BinaryType {
+export class BinaryType {
   public static MIN_UNSIGNED_VALUE: number = 0
   /**
    * The unsigned integer representation of the type as a number (IEEE double precision floating point).
    * Has to be always positive use toSignedInteger method to get interpretation of negative values
    */
-  abstract readonly value: number
-  abstract readonly numberOfBitsForType: number
-  abstract readonly maxValueForType: number
+  public readonly value: number
+  public readonly numberOfBitsForType: number
+  public readonly maxValueForType: number
 
-  constructor() {
-    //since value here is sometimes undefined, abstract properties are used
+  constructor(value: number, numberOfBitsForType: number) {
+    this.value = value
+    this.numberOfBitsForType = numberOfBitsForType
+    this.maxValueForType = Math.pow(2, numberOfBitsForType) - 1
   }
```

Comment:
There was a discussion between you and @clooudch 
I guess this was the consense of it.